src/ngx_http_lua_shdict.c

```diff
@@ -2843,6 +2843,124 @@ ngx_http_lua_ffi_shdict_flush_all(ngx_shm_zone_t *zone)
 
     return NGX_OK;
 }
+
+
+static ngx_int_t
+ngx_http_lua_shdict_peek(ngx_shm_zone_t *shm_zone, ngx_uint_t hash,
```

Comment:
The idea with this `peek` function is to not consider the node "touched" and avoid the node being put at the queue's head. Open to feedback.